src/main/java/it/smartcommunitylab/aac/webauthn/provider/WebAuthnAuthenticationProvider.java

```diff
@@ -0,0 +1,114 @@
+package it.smartcommunitylab.aac.webauthn.provider;
+
+import java.time.Instant;
+
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.util.Assert;
+
+import it.smartcommunitylab.aac.SystemKeys;
+import it.smartcommunitylab.aac.common.NoSuchUserException;
+import it.smartcommunitylab.aac.core.auth.ExtendedAuthenticationProvider;
+import it.smartcommunitylab.aac.core.auth.UserAuthenticatedPrincipal;
+import it.smartcommunitylab.aac.webauthn.auth.RegisterCredentialAuthenticationToken;
+import it.smartcommunitylab.aac.webauthn.auth.WebAuthnAuthenticationException;
+import it.smartcommunitylab.aac.webauthn.model.WebAuthnUserAuthenticatedPrincipal;
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnUserAccount;
+import it.smartcommunitylab.aac.webauthn.service.WebAuthnUserAccountService;
+import it.smartcommunitylab.aac.webauthn.service.WebAuthnUserDetailsService;
+
+public class WebAuthnAuthenticationProvider extends ExtendedAuthenticationProvider {
+
+    private final WebAuthnIdentityProviderConfigMap config;
+
+    private final WebAuthnUserAccountService userAccountService;
+    private final WebAuthnUserDetailsService userDetailsService;
+    private final DaoAuthenticationProvider authProvider;
+
+    public WebAuthnAuthenticationProvider(String providerId,
+            WebAuthnUserAccountService userAccountService,
+            WebAuthnAccountService accountService,
+            WebAuthnCredentialsService credentialsService,
+            WebAuthnIdentityProviderConfig providerConfig, String realm) {
+        super(SystemKeys.AUTHORITY_WEBAUTHN, providerId, realm);
+        Assert.notNull(userAccountService, "user account service is mandatory");
+        Assert.notNull(providerConfig, "provider config is mandatory");
+        this.config = providerConfig.getConfigMap();
+        this.userAccountService = userAccountService;
+
+        // build a userDetails service
+        userDetailsService = new WebAuthnUserDetailsService(userAccountService, realm);
+
+        // build our WebAuthn auth provider by wrapping spring dao authprovider
+        authProvider = new DaoAuthenticationProvider();
```

Comment:
why using daoauthprovider when we don't have a password to compare?

Reply:
Removing it